### PR TITLE
Support core.query_model()

### PR DESCRIPTION
**Pull request by hegdeadithyak**
### Details:
Added Query Model to C++ Core Class and Updated TS addons

Solves : #24373

### CODE REVIEW

**Comment by hegdeadithyak** [comment]
@vishniakov-nikolai  I need some help with writing Tests for this.

File: src/bindings/js/node/src/core_wrap.cpp [hunk 1]
```diff
@@ -375,3 +376,33 @@ void CoreWrap::add_extension(const Napi::CallbackInfo& info) {
         reportError(info.Env(), err.what());
     }
 }
+
+Napi::Value CoreWrap::query_model(const Napi::CallbackInfo& info) {
+    try {
+        if (info.Length() < 2 || !info[0].IsObject() || !info[1].IsString()) {
```
Comment (almilosz) [inline src/bindings/js/node/src/core_wrap.cpp:382]: You should check if info[0] contain model object like this:
`
const auto& model_prototype = info.Env().GetInstanceData<AddonData>()->model;
 if (model_prototype && model.InstanceOf(model_prototype.Value().As<Napi::Function>())) { }
`
The exact form of this condition I leave to you ;)

File: src/bindings/js/node/include/core_wrap.hpp [hunk 2]
```diff
@@ -68,7 +68,17 @@ class CoreWrap : public Napi::ObjectWrap<CoreWrap> {
     Napi::Value set_property(const Napi::CallbackInfo& info);
     Napi::Value get_property(const Napi::CallbackInfo& info);
 
+    /** @brief Query device if it supports the specified model with specified properties.
```
Comment (almilosz) [inline src/bindings/js/node/include/core_wrap.hpp:71]: Move this description to addon.ts
You will have to rewrite it according to [typedoc](https://typedoc.org/example/)
You can look at how other methods in addon.ts are described

**Comment by almilosz** [comment]
> @vishniakov-nikolai I need some help with writing Tests for this.

The test should be put in this file: `src/bindings/js/node/tests/core.test.js`
You can use model from here -> `src/bindings/js/node/tests/test_models`
To test the functionality you can check if each operation in the output of the method is supported by the selected device
Something like this:
```
query = core.query_model(model=model, device_name=device)
assert device in next(iter(set(query.values()))), "Wrong device for some layers"
```

**Comment by vishniakov-nikolai** [comment]
Hi @hegdeadithyak! To prevent close of this PR, please add fresh commit to it or let us know if you need more time. Thank you!

**Pull request closed without merging**